test/gurka/test_reproduce_tile_build.cc

```diff
@@ -0,0 +1,99 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+using namespace valhalla::baldr;
+
+void assert_tile_equalish(const GraphTile& a, const GraphTile& b) {
+  // expected size
+  ASSERT_EQ(a.header()->end_offset(), b.header()->end_offset());
+
+  // check the first chunk after the header
+  ASSERT_EQ(memcmp(reinterpret_cast<const char*>(a.header()) + sizeof(GraphTileHeader),
+                   reinterpret_cast<const char*>(b.header()) + sizeof(GraphTileHeader),
+                   (reinterpret_cast<const char*>(b.GetBin(0, 0).begin()) -
+                    reinterpret_cast<const char*>(b.header())) -
+                       sizeof(GraphTileHeader)),
+            0);
+
+  // check the stuff after the bins
+  ASSERT_EQ(memcmp(reinterpret_cast<const char*>(a.header()) + a.header()->edgeinfo_offset(),
+                   reinterpret_cast<const char*>(b.header()) + b.header()->edgeinfo_offset(),
+                   b.header()->end_offset() - b.header()->edgeinfo_offset()),
+            0);
+
+  // if the header is as expected
+  const auto *ah = a.header(), *bh = b.header();
+  if (ah->access_restriction_count() == bh->access_restriction_count() &&
```

Comment:
The boolean expression in this if statement will be clearer factored into a variable, or possibly a standalone function.

Reply:
I also noticed this giant expression.. 
The first thought were replace it with something like
```cpp
// before
if (a && b && c && d....) { 

} else {
 Fail();
}

// after
EXPECT_TRUE(a);
EXPECT_TRUE(b);
EXPECT_TRUE(c);

// gtest supports custom messages if the check fails
// might be an option not to create new variables and provide additional context at the same time
EXPECT_EQ(a.edges, b.edges) << "Edges differ";

```

Reply:
To be honest that was copied from another non-gurka [test](https://github.com/valhalla/valhalla/blob/6387824f6258e2c434eb297178df01cd86a9aff8/test/graphtilebuilder.cc#L30). If it's fine to reuse code between gurka and non-gurka tests - I can easily extract it into header and use in both test

Reply:
> The boolean expression in this if statement will be clearer factored into a variable, or possibly a standalone function.

Good point. Done

Reply:
Yeah, moving the if-block to a new new gtest assertion is A+.